File: 240429/최대 H 점수 2/maximum-h-score-2.py
N, L = map(int, input().split())
seqs = list(map(int, input().split()))

for H in range(N, -1, -1):
    l_cnt, h_cnt = 0, 0
    for i in range(N):
        if seqs[i] >= H:
            h_cnt += 1
        elif seqs[i] == H - 1:
            h_cnt += 1
            l_cnt += 1

    if h_cnt >= H and (l_cnt - (h_cnt - H)) <= L:
        print(H)
        break

File: 260331/최대 H 점수 2/maximum-h-score-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, L;
    cin >> N >> L;
    vector<int>seqs(N);

    for (int i = 0; i < N; i++) 
    {
        cin >> seqs[i];
    }

    // Please write your code here.
    for(int H = N; H >= 0; H--)
    {
        int l_cnt = 0, h_cnt = 0;
        for(int i = 0; i < N; i++)
        {
            if(seqs[i] >= H)
            {
                h_cnt += 1;
            }
            else if(seqs[i] == H - 1)
            {
                h_cnt += 1;
                l_cnt += 1;
            }
        }

        if(h_cnt >= H && (l_cnt - (h_cnt - H)) <= L)
        {
            cout << H << "\n";
            break;
        }
    }
    return 0;
}